### ISSUE TITLE
Unable to Pass Down Parameters

### ISSUE THREAD

**Issue by brandon-james105** [chris579/MvvmBlazor#11]
I am unable to pass a parameter down from the page to the ViewModel. The BindingContext is not created until after the page is rendered so I cannot set it in the component constructor.

**Comment by chris579** [comment]
Can you show me some example code so I can get an idea what you're doing? Much appreciated!

**Comment by brandon-james105** [comment]
This [parameter](https://github.com/brandon-james105/BlazorPlanningPoker/blob/master/PlanningPoker/Views/SessionBase.cs#L14) cannot be put into the viewmodel in the constructor.

**Comment by chris579** [comment]
I see, I was also facing this issue last week. I need to find a proper solution for that. The problem is that the init of the BindingContext happens when OnInitialized is triggered. Therefore you can't just pass the parameter down.

From my current standpoint I would create a logic that passes down parameters on it's own. Imagine that you define your parameter in the component

```csharp
[Parameter]
public string SessionId { get; set; }
```

and define the same parameter in your view model 
```csharp
public class ViewModel: ViewModelBase
{
    [Parameter]
    public string SessionId { get; set; }
}
```

and some logic would take care about that parameter and passes it down to the view model.

**Comment by brandon-james105** [comment]
I was passing it down before using your library by overriding the `OnInitialized` / `OnInitializedAsync `method in the Razor Component. The objects aren't instantiated in the constructor.

**Comment by chris579** [comment]
You're not supposed to override `OnInitialized` / `OnInitializedAsync`. The goal is to keep them away from the user, they are marked as `sealed` for this reason. I will implement my approach and look how it feels.

**Comment by chris579** [comment]
This problem is addressed in `v1.1.4` which has been pushed some seconds ago and is currently validated by NuGet.

**Comment by brandon-james105** [comment]
Thank you I confirm this is working in my project now
